examples/platforms/posix/ble_nimble.c

```diff
@@ -0,0 +1,1111 @@
+/*
+ *
+ *    Copyright (c) 2018 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    This document is the property of Nest. It is considered
+ *    confidential and proprietary information.
+ *
+ *    This document may not be reproduced or transmitted in any form,
+ *    in whole or in part, without the express written permission of
+ *    Nest.
+ *
+ */
+
+/**
+ *    @file
+ *      This file implements OpenThread BLE platform interface
+ *      that maps into nlplatform port of NimBLE API.
+ *
+ */
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <common/logging.hpp>
+#include <openthread/platform/ble.h>
+#include <openthread/platform/ble_hci.h>
+
+#include "ble_nimble_event.h"
+
+#include <nimble/nimble_npl.h>
+#include <nimble/nimble_port.h>
+
+#include "ble_l2cap_priv.h"
+#include <host/ble_hs.h>
+#include <nimble/ble.h>
+#include <services/ans/ble_svc_ans.h>
+#include <services/gap/ble_svc_gap.h>
+#include <services/gatt/ble_svc_gatt.h>
+
+#define TASK_DEFAULT_PRIORITY 1
+#define TASK_DEFAULT_STACK NULL
+#define TASK_DEFAULT_STACK_SIZE 400
+
+/// Max amount of time in [ms] to scan on connection request.
+#define DEFAULT_CONN_DISC_INTERVAL 1000
+#define DEFAULT_ADDR_TYPE BLE_OWN_ADDR_RANDOM
+
+#define BLE_MAX_NUM_SERVICES 2
```

Comment:
nit: In nrf52840 we have created platform-config.h file, what about to move platform specific config there?